kimchi/src/linearization.rs

```diff
@@ -30,15 +31,24 @@ use crate::circuits::{
 };
 use ark_ff::{FftField, PrimeField, SquareRootField, Zero};
 
-/// Get the expresion of constraints.
+/// Get the expression of all constraints.
+///
+/// If the `use_feature_flags` argument is `false`, this will generate an expression using the
+/// `Expr::IfFeature` variant for each of the flags.
 ///
 /// # Panics
 ///
 /// Will panic if `generic_gate` is not associate with `alpha^0`.
 pub fn constraints_expr<F: PrimeField + SquareRootField>(
-    feature_flags: Option<&FeatureFlags>,
+    cs: &ConstraintSystem<F>,
+    use_feature_flags: bool,
     generic: bool,
 ) -> (Expr<ConstantExpr<F>>, Alphas<F>) {
+    let feature_flags = if use_feature_flags {
+        Some(cs.feature_flags)
+    } else {
+        None
+    };
     // register powers of alpha so that we don't reuse them across mutually inclusive constraints
     let mut powers_of_alpha = Alphas::<F>::default();
 
```

Comment:
here, instead of registering VarbaseMul constraints, use the maximum so it is more dynamic.